Objects/weakrefobject.c

```diff
@@ -763,7 +748,7 @@ _PyWeakref_ProxyType = {
     &proxy_as_number,                   /* tp_as_number */
     &proxy_as_sequence,                 /* tp_as_sequence */
     &proxy_as_mapping,                  /* tp_as_mapping */
-    proxy_hash,                         /* tp_hash */
+    0,                                  /* tp_hash */
```

Comment:
Perhaps add a comment here that tp_hash is intentionally omitted.

Reply:
:+1:

Reply:
```suggestion
// Notice that tp_hash is intentionally omitted as proxies are "mutable" (when the reference dies).
    0,                                  /* tp_hash */
```